src/Modules/Data/Commands.Data/GetPowerBIDatasource.cs

```diff
@@ -0,0 +1,105 @@
+    {
+        public const string CmdletVerb = VerbsCommon.Get;
+        public const string CmdletName = "PowerBIDatasource";
+
+        #region ParameterSets
+        private const string IdParameterSetName = "Id";
+        private const string NameParameterSetName = "Name";
+        private const string ListParameterSetName = "List";
+        private const string ObjectIdParameterSetName = "ObjectAndId";
+        private const string ObjectNameParameterSetName = "ObjectAndName";
+        private const string ObjectListParameterSetName = "ObjectAndList";
+        #endregion
+
+        
+        #region Parameters
+        [Alias("DatasetKey")]
+        [Parameter(Mandatory = true, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = IdParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSetName)]
+        public Guid DatasetId { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ObjectIdParameterSetName, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectNameParameterSetName, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectListParameterSetName, ValueFromPipeline = true)]
+        public Dataset Dataset { get; set; }
+
+        [Alias("GroupId")]
+        [Parameter(Mandatory = false)]
+        public Guid WorkspaceId { get; set; }
+
+        [Alias("DatasourceId")]
+        [Parameter(Mandatory = true, ParameterSetName = IdParameterSetName)]
+        public Guid Id { get; set; }
+
+        [Alias("DatasoureName")]
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSetName)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public PowerBIUserScope Scope { get; set; } = PowerBIUserScope.Individual;
+        #endregion
+
+        #region Constructors
+        public GetPowerBIDatasource() : base() { }
+
+        public GetPowerBIDatasource(IPowerBIClientCmdletInitFactory init) : base(init) { }
+        #endregion
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+            if (this.Scope == PowerBIUserScope.Organization && this.WorkspaceId != default)
+            {
+                this.Logger.ThrowTerminatingError($"{nameof(this.WorkspaceId)} is only applied when -{nameof(this.Scope)} is set to {nameof(PowerBIUserScope.Individual)}");
+            }
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if(this.Dataset != null)
+            {
+                this.DatasetId = this.Dataset.Id;
+            }
+
+            IEnumerable<Datasource> datasources = null;
+            using (var client = this.CreateClient())
+            {
+                datasources = this.Scope == PowerBIUserScope.Individual ?
+                    client.Datasets.GetDatasources(this.DatasetId, this.WorkspaceId) :
```

Comment:
You should also be able to return datasources without workspace ID specified for the Individual scope.

Reply:
If we can, it's in the service code but not documented in the SDK\API - https://docs.microsoft.com/en-us/rest/api/power-bi/datasets/getdatasources